File: examples/gripper.cpp
#include <iostream>
#include "rbpodo/rbpodo.hpp"

using namespace std::chrono_literals;
using namespace rb;

int main() {
  try {
    // Make connection
    auto robot = podo::Cobot("10.0.2.7");
    auto rc = podo::ResponseCollector();

        robot.gripper_rts_rhp12rn_select_mode(rc, podo::GripperConnectionPoint::ToolFlange_Advanced, true);
        std::this_thread::sleep_for(0.5s);

        robot.gripper_rts_rhp12rn_force_control(rc, podo::GripperConnectionPoint::ToolFlange_Advanced, 100);
        std::this_thread::sleep_for(2s);

        robot.gripper_rts_rhp12rn_force_control(rc, podo::GripperConnectionPoint::ToolFlange_Advanced, -100);
        std::this_thread::sleep_for(2s);

        robot.gripper_rts_rhp12rn_force_control(rc, podo::GripperConnectionPoint::ToolFlange_Advanced, 0);
  } catch (const std::exception& e) {
    std::cerr << e.what() << std::endl;
    return 1;
  }
  return 0;
}

File: examples/gripper.py
import time
import rbpodo as rb

robot = rb.Cobot("10.0.2.7")
rc = rb.ResponseCollector()

robot.gripper_rts_rhp12rn_select_mode(rc, rb.GripperConnectionPoint.ToolFlange_Advanced, True)
time.sleep(0.5)

robot.gripper_rts_rhp12rn_force_control(rc, rb.GripperConnectionPoint.ToolFlange_Advanced, 100)
time.sleep(2)

robot.gripper_rts_rhp12rn_force_control(rc, rb.GripperConnectionPoint.ToolFlange_Advanced, -100)
time.sleep(2)

robot.gripper_rts_rhp12rn_force_control(rc, rb.GripperConnectionPoint.ToolFlange_Advanced, 0)
rc = rc.error().throw_if_not_empty()
